onchain/permissionless-arbitration/src/Match.sol

```diff
@@ -111,7 +112,7 @@ library Match {
         state.leftNode = newLeftNode;
         state.rightNode = newRightNode;
 
-        state.runningLeafPosition += uint64(1 << state.currentHeight); // TODO: verify
+        state.runningLeafPosition += 1 << state.currentHeight; // TODO: verify
```

Comment:
Ultimately with tests haha, but if you can reason about it now that's also ok (we'll still need test).

These are lines I believe could have an off-by-one error. They make sense to me as they are, but I wouldn't trust them until I saw some tests.

Reply:
I think we can write Solidity tests to verify this. Ultimately the integration tests with real `step` and `node` can also verify it. Although it will be harder to debug with all the components in place.

Reply:
Yeah, I'm making all these changes to run a test using the Lua prototype. It should catch some bugs, but we'll need good unit tests too. Since the code is still a bit brittle/unstable, I'm postponing deeper tests.